includes/data-stores/class-wc-product-data-store-posts.php

```diff
@@ -0,0 +1,142 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC Product Data Store.
+ *
+ * @version  2.7.0
+ * @category Class
+ * @author   WooThemes
+ */
+class WC_Product_Data_Store_Posts extends WC_Data_Store_Posts implements WC_Object_Data_Store {
+
+	/**
+	 * Stores a new product in the database.
+	 * @param WC_Product Object
+	 */
+	public function create( $product ) {
+
+	}
+
+	/**
+	 * Read a product from the database.
+	 * @param WC_Product
+	 */
+	public function read( &$product ) {
+		$product->set_defaults();
+
+		if ( ! $product->get_id() || ! ( $post_object = get_post( $product->get_id() ) ) ) {
+			// @todo throw an error? set to empty object? null?
```

Comment:
Not sure if exception is appropriate unless we catch it in wc_get_product? Might be ok - we throw exceptions in the variation class right now if the parent is invalid.